MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/PropertyIconDeterminationActivity.kt

```diff
@@ -0,0 +1,161 @@
+package com.mapbox.mapboxandroiddemo.examples.dds
+
+import android.graphics.BitmapFactory
+import com.mapbox.mapboxandroiddemo.R
+import android.graphics.PointF
+import android.os.Bundle
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.FeatureCollection
+import com.mapbox.geojson.Point
+import com.mapbox.mapboxsdk.Mapbox
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.style.expressions.Expression.*
+import com.mapbox.mapboxsdk.style.layers.Property
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory.*
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource
+
+
+class PropertyIconDeterminationActivity : AppCompatActivity(),  MapboxMap.OnMapClickListener {
+
+
+    private val SOURCE_ID = "SOURCE_ID"
+    private val RED_ICON_ID = "RED_ICON_ID"
+    private val YELLOW_ICON_ID = "YELLOW_ICON_ID"
+    private val LAYER_ID = "LAYER_ID"
+    private val ICON_PROPERTY = "ICON_PROPERTY"
+    private var mapView: MapView? = null
```

Comment:
You don';t need to rely on nullable views and using findViewById is you use android-kotlin-extensions: https://antonioleiva.com/kotlin-android-extensions/

Reply:
kotlinx.android.synthetic is no longer a recommended practice. It is deprecated and you should use ViewBinding instead:
* https://android-developers.googleblog.com/2020/11/the-future-of-kotlin-android-extensions.html
* https://proandroiddev.com/migrating-the-deprecated-kotlin-android-extensions-compiler-plugin-to-viewbinding-d234c691dec7